### PR TITLE
Algorithm/curriculum learning

### PR DESCRIPTION
**Pull request by JeremyFongSP**
Merge Curriculum Training to master branch

### CODE REVIEW

File: client/curriculum/curriculum.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+
+def curriculum_learning(obj, file):
+    """Curriculum training for better generalization, currently implemented in
+    an ad-hoc method. This only serves as proof of concept"""
+
+    with open(file, 'r') as curr_file:
+        yaml_list = yaml.safe_load(curr_file)
+
+    curriculum_training_set = yaml_list['curr_list']
```
Comment (jaberkow) [inline client/curriculum/curriculum.py:14]: Add a comment here explaining what `curriculum_training_set` is.

File: client/curriculum/curriculum.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
+from random import random
+import yaml
+
+
+def curriculum_learning(obj, file):
+    """Curriculum training for better generalization, currently implemented in
```
Comment (jaberkow) [inline client/curriculum/curriculum.py:6]: Disregard this comment if the function is going to change significantly for the final implementation, but the `file` and `obj` arguments should be better documented and defined.

**Review by jaberkow — APPROVED** [review]
Your code looks, for the most part, very clean and professional.  It's clear to me that you've taken the time to understand the structure of the carla codebase very thoroughly.

**Comment by jaberkow** [comment]
Resolve the conflict in `vpy_carla.py` (which seems to be a sign difference) and I will approve merging this PR.

File: client/curriculum/curriculum.py (hunk 2 above)
Comment (JeremyFongSP) [reply client/curriculum/curriculum.py:6]: I've added some comments. Please let me know if you think this is enough or if I should elaborate further! Thanks for the great feedback.

**Pull request merged**